packages/backend/server/src/modules/quota/storage.ts

```diff
@@ -0,0 +1,64 @@
+import type { Storage } from '@affine/storage';
+import {
+  ForbiddenException,
+  Inject,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+
+import { StorageProvide } from '../../storage';
+import { PermissionService } from '../workspaces/permission';
+import { QuotaService } from './quota';
+
+@Injectable()
+export class QuotaManagementService {
+  constructor(
+    private readonly quota: QuotaService,
+    private readonly permissions: PermissionService,
+    @Inject(StorageProvide) private readonly storage: Storage
+  ) {}
+
+  async getUserQuota(userId: string) {
+    const quota = await this.quota.getUserQuota(userId);
+    if (quota) {
+      return {
+        name: quota.feature.feature,
+        reason: quota.reason,
+        createAt: quota.createdAt,
+        expiredAt: quota.expiredAt,
+        blobLimit: quota.feature.configs.blobLimit,
+        storageQuota: quota.feature.configs.storageQuota,
+      };
+    }
+    return null;
+  }
+
+  // TODO: lazy calc, need to be optimized with cache
+  async getUserUsage(userId: string) {
```

Comment:
if we record the inform in database, we may commit the change frequently
but maybe we can consider storing the usage data in redis and updating it when it changes to avoid frequent database query
i will use redis store the usage in next pr in the tree

Reply:
maybe we could make a `@Cache` interceptor to have an automatic cache convention

Reply:
done in #5178